org-ql.el

```diff
@@ -60,6 +60,11 @@
 Like `org-clock-line-re', but matches the timestamp range in a
 match group.")
 
+(defconst org-ql-log-done-regexp
+  (rx bol "- State \"DONE\"" (1+ blank) "from \"WAITING\"" (1+ blank) (group (1+ not-newline)))
+  ;; TODO: Remove hardcoded states, i.e DONE & WAITING
```

Comment:
This todo should be fixed before merging.  So the regexp, rather than matching DONE from WAITING, should probably match an arbitrary state from an arbitrary state, so the predicate could work like `(logbook-state "DONE")` or `(logbook-state "DONE" "WAITING")`.

Reply:
Can you help me with this @alphapapa ? 
I've tried adding these arguments to the `predicate`, but don't know how to change the `preambles` and `normalizers` part.

Reply:
Those are `pcase` forms.  Pcase isn't the easiest tool to learn, but it's very powerful.  I'll take a look at this when I get time, but that may not be for a while, so don't be afraid to dive in to Pcase's documentation in the meantime.